tests/python/relax/test_op_binary.py

```diff
@@ -0,0 +1,135 @@
+    _check_inference(bb, relax.op.less(x, y0), relax.TensorStructInfo((2, 3), "bool"))
+    _check_inference(bb, relax.op.less(x, y1), relax.TensorStructInfo((2, 3), "bool"))
+
+
+def test_infer_struct_info_binary_symbolic():
+    bb = relax.BlockBuilder()
+    m = tir.Var("m", "int64")
+    n = tir.Var("n", "int64")
+    k = tir.Var("k", "int64")
+    x0 = relax.Var("x", R.Tensor((m, n), "float32"))
+    x1 = relax.Var("x", R.Tensor((1, n), "float32"))
+    x2 = relax.Var("x", R.Tensor((k, n, m), "float32"))
+    x3 = relax.Var("x", R.Tensor((3, 1, n), "float32"))
+    x4 = relax.Var("x", R.Tensor("float32", ndim=2))
+    y0 = relax.Var("y", R.Tensor((m, n), "float32"))
+    y1 = relax.Var("y", R.Tensor((m, n + 2), "float32"))
+    y2 = relax.Var("y", R.Tensor((4, k, m, 1), "float32"))
+    y3 = relax.Var("y", R.Tensor("float32", ndim=2))
+    y4 = relax.Var("y", R.Tensor("float32", ndim=-1))
+    _check_inference(bb, relax.op.add(x0, y0), relax.TensorStructInfo((m, n), "float32"))
+    _check_inference(bb, relax.op.add(x0, y1), relax.TensorStructInfo(dtype="float32", ndim=2))
+    _check_inference(bb, relax.op.subtract(x1, y0), relax.TensorStructInfo((m, n), "float32"))
+    _check_inference(bb, relax.op.multiply(x1, y2), relax.TensorStructInfo((4, k, m, n), "float32"))
+    _check_inference(bb, relax.op.divide(x2, y2), relax.TensorStructInfo(dtype="float32", ndim=4))
+    _check_inference(
+        bb, relax.op.floor_divide(x2, y3), relax.TensorStructInfo(dtype="float32", ndim=3)
+    )
+    _check_inference(bb, relax.op.add(x3, y3), relax.TensorStructInfo(dtype="float32", ndim=3))
+    _check_inference(bb, relax.op.add(x4, y0), relax.TensorStructInfo(dtype="float32", ndim=2))
+    _check_inference(bb, relax.op.add(x4, y2), relax.TensorStructInfo(dtype="float32", ndim=4))
+    _check_inference(bb, relax.op.add(x4, y3), relax.TensorStructInfo(dtype="float32", ndim=2))
+    _check_inference(bb, relax.op.add(x4, y4), relax.TensorStructInfo(dtype="float32", ndim=-1))
+
+
+def test_infer_struct_info_binary_shape_unequal_const_int():
+    bb = relax.BlockBuilder()
+    x0 = relax.Var("x", R.Tensor((2, 3), "float32"))
+    y0 = relax.Var("y", R.Tensor((2, 4), "float32"))
+    with pytest.raises(TVMError):
+        bb.normalize(relax.op.add(x0, y0))
+
+
+def test_infer_struct_info_binary_arith_dtype_mismatch():
+    bb = relax.BlockBuilder()
+    x = relax.Var("x", R.Tensor((2, 3), "float32"))
+    y = relax.Var("y", R.Tensor((2, 3), "int32"))
+    with pytest.raises(TVMError):
+        bb.normalize(relax.op.add(x, y))
+
+
+def test_infer_struct_info_binary_wrong_input():
+    bb = relax.BlockBuilder()
+    x0 = relax.Var("x", relax.ShapeStructInfo((2, 3)))
+    x1 = relax.Var("x", relax.FuncStructInfo([], R.Tensor((2, 3), "float32")))
+    y = relax.Var("y", R.Tensor((2, 3), "float32"))
+
+    with pytest.raises(TVMError):
+        bb.normalize(relax.op.add(x0, y))
+    with pytest.raises(TVMError):
```

Comment:
add negative cases:

- wrong number of inputs (but previous two inputs are good)
- pass in non-tensor types into the data